alicat/util.py

```diff
@@ -75,7 +104,8 @@ async def _handle_communication(self, command: str) -> str | None:
         """Manage communication, including timeouts and logging."""
         try:
             await self._write(command)
-            result = await self._readline()
+            future = self._readline()
```

Comment:
was there a problem with `await asyncio.wait_for(self._readline(), timeout))`?

Reply:
No, actually I'll revise to that. Good catch.  The previous implementation had two "repeats" of this same code block and one was written this way, while the other had it all in-line.

Reply:
(eventually) I got around to re-analyzing this and found a lot of needless redundancies eg. https://github.com/alexrudd2/alicat/commit/2dd5832935bfb967b70ff552413fb54bbffc9ac6

 Thanks very much for pointing it out @griffinmilsap